tests/suite/snippets_filter_test.go

```diff
@@ -0,0 +1,340 @@
+		})
+
+		Specify("snippetsFilters are accepted", func() {
+			snippetsFilterNames := []string{
+				"all-contexts",
+				"grpc-all-contexts",
+			}
+
+			for _, name := range snippetsFilterNames {
+				nsname := types.NamespacedName{Name: name, Namespace: namespace}
+
+				err := waitForSnippetsFilterToBeAccepted(nsname)
+				Expect(err).ToNot(HaveOccurred(), fmt.Sprintf("%s was not accepted", name))
+			}
+		})
+
+		Context("verify working traffic", func() {
+			It("should return a 200 response for HTTPRoute", func() {
+				Eventually(
+					func() error {
+						return expectRequestToSucceed(baseURL, address, "URI: /coffee")
+					}).
+					WithTimeout(timeoutConfig.CreateTimeout).
+					WithPolling(5 * time.Second).
+					Should(Succeed())
+			})
+		})
+
+		Context("nginx directives", func() {
+			var conf *framework.Payload
+			snippetsFilterPrefix := "SnippetsFilter_"
+			filePrefix := "/etc/nginx/includes/"
+			httpRoutePrefix := fmt.Sprintf("%s_all-contexts.conf", namespace)
+			grpcRoutePrefix := fmt.Sprintf("%s_grpc-all-contexts.conf", namespace)
+
+			BeforeAll(func() {
+				podNames, err := framework.GetReadyNGFPodNames(k8sClient, ngfNamespace, releaseName, timeoutConfig.GetTimeout)
+				Expect(err).ToNot(HaveOccurred())
+				Expect(podNames).To(HaveLen(1))
+
+				ngfPodName := podNames[0]
+
+				conf, err = resourceManager.GetNginxConfig(ngfPodName, ngfNamespace)
+				Expect(err).ToNot(HaveOccurred())
+			})
+
+			DescribeTable("are set properly for",
+				func(expCfgs []framework.ExpectedNginxField) {
+					for _, expCfg := range expCfgs {
+						Expect(framework.ValidateNginxFieldExists(conf, expCfg)).To(Succeed())
+					}
+				},
+				Entry("HTTPRoute", []framework.ExpectedNginxField{
+					{
+						Directive: "aio",
+						Value:     "on",
+						File:      fmt.Sprintf("%s%shttp_%s", filePrefix, snippetsFilterPrefix, httpRoutePrefix),
+					},
+					{
+						Directive: "auth_delay",
+						Value:     "10s",
+						File:      fmt.Sprintf("%s%shttp.server_%s", filePrefix, snippetsFilterPrefix, httpRoutePrefix),
+					},
+					{
+						Directive: "keepalive_time",
+						Value:     "10s",
+						File:      fmt.Sprintf("%s%shttp.server.location_%s", filePrefix, snippetsFilterPrefix, httpRoutePrefix),
+					},
+					{
+						Directive: "worker_priority",
+						Value:     "0",
+						File:      fmt.Sprintf("%s%smain_%s", filePrefix, snippetsFilterPrefix, httpRoutePrefix),
+					},
+					{
+						Directive: "include",
+						Value:     fmt.Sprintf("%s%shttp.server_%s", filePrefix, snippetsFilterPrefix, httpRoutePrefix),
+						Servers:   []string{"cafe.example.com"},
+						File:      "http.conf",
+					},
+					{
+						Directive: "include",
+						Value:     fmt.Sprintf("%s%shttp.server.location_%s", filePrefix, snippetsFilterPrefix, httpRoutePrefix),
+						File:      "http.conf",
+						Location:  "/coffee",
+						Servers:   []string{"cafe.example.com"},
+					},
+				}),
```

Comment:
Doesn't look like we're checking for the `include` directives for the `http` and `main` files.